botocore/utils.py

```diff
@@ -308,6 +343,11 @@ def _fetch_metadata_token(self):
                     raise BadIMDSRequestError(request)
             except ReadTimeoutError:
                 return None
+            except HTTPClientError as e:
+                if isinstance(e.kwargs.get('error'), LocationParseError):
+                    raise InvalidIMDSEndpointError(endpoint=url, error=e)
+                else:
+                    raise HTTPClientError(error=e.kwargs.get('error'))
```

Comment:
You should be able to just do `raise` or `raise e` here and it will rethrow the current exception being caught. That avoids having to try and recreate it again.

i.e.
```python
except HTTPClientError as e:
    if isinstance(e.kwargs.get('error'), LocationParseError):
        raise InvalidIMDSEndpointError(endpoint=url, error=e)
    raise
```